scipion/install/funcs.py

```diff
@@ -296,30 +296,31 @@ def getPythonPackagesFolder():
 
     @staticmethod
     def getIncludeFolder():
-        return '%s/include' % (Environment.getSoftware())
+        return Environment.getSoftware('include')
 
     def getLib(self, name):
-        return '%s/lib%s.%s' % (Environment.getLibFolder(),
-                                name, self._libSuffix)
+
+        return Environment.getLibFolder('lib%s.%s' % (name, self._libSuffix))
 
     @staticmethod
-    def getBinFolder():
-        return '%s/bin' % Environment.getSoftware()
+    def getBinFolder(*paths):
+        return  os.path.join(mkdir(Environment.getSoftware('bin')), *paths)
 
     @staticmethod
     def getBin(name):
-        return '%s/%s' % (Environment.getBinFolder(), name)
+        return  Environment.getBinFolder(name)
 
     @staticmethod
     def getTmpFolder():
-        return '%s/tmp' % Environment.getSoftware()
+        return  mkdir(Environment.getSoftware('tmp'))
```

Comment:
Every time the getTmpFolder() is called there will be a mkdir command?

Reply:
mkdir is a function defined below where only create the dir if it doesn't exist. Maybe the function name is quite confusing...

Reply:
Still, it checks if the folder exists every time you call the function, anyway I think these functions are not called too much

Reply:
I know, I do not like this much, but as you said, @delarosatrevin this code is rarely invoked. Only at installation time.

The reason for this is that the config was doing this before (mandatory). Now being it optional, we are missing that "step" to prepare the folder structure.